src/libAtomVM/bitstring.c

```diff
@@ -119,10 +140,64 @@ bool bitstring_utf8_encode(avm_int_t c, uint8_t *buf, size_t *out_size)
     return true;
 }
 
+bool bitstring_utf8_decode(int32_t *c, uint8_t *buf, size_t len, size_t *out_size)
```

Comment:
Let's use `const` if we are not going to change memory. Same comment applies also to the other functions here below.

Reply:
We do modify *c in the decode functions.  Do you mean the encode functions?  Should I change those?

Reply:
@bettio pointed out that `buf` can be const.  Fixed.